### DISCUSSION TITLE
Context Attachment Should be Configurable by Chat Participants

### DISCUSSION THREAD

**Discussion by amitjoshi438** [microsoft/vscode-discussions#1262] [Extension Development QnA]
Currently even when chat participant is not using the context, the option to attach context is there. I think it can be misleading for users as users would believe that attached context will be used but participant is not actually referring to it.

If we can have some control where after using @participant, based on configs attachments option can be enabled/disabled.

cc: @isidorn @roblourens

**Comment by amitjoshi438** [comment]
Also, since the chat variables are global it does help in getting the context but if not used by participants adds handling overhead to remove the #variable from the prompt, again can become confusing for user.

**Comment by isidorn** [comment]
Sorry about the slow response, I was on vacation.

What is your exact proposal here? I agree that it can be confusing to the user, as VS Code does not really know if a participant uses context or not.

**Reply by amitjoshi438** [reply]
Hey  @isidorn , thanks for responding. If possible, it will be good to make this OOB chat variables configurable for a @participant. If a @participant wants to leverage them, they can enable them based on requirement and other variables should not be shown if @participant is used.

**Reply by amitjoshi438** [reply]
Hi @isidorn, let me know if there are any plans to update this experience.

**Reply by isidorn** [reply]
No plans updating the experience, but I still do not fully understand what you are trying to achieve here.
Can you share an example? How it behaves now, and what would you like to be different. Thanks

**Reply by amitjoshi438** [reply]
**Current experience:**
Users Prompt - 
@myParticipant Add one more section to this webpage #file (attached file)

If @myParticipant doesn't have any provision to use #file information passed by users. It needs to be removed from the prompt before passing to LLM but user will be confused and may think that @myParticipant is not working properly.

**Suggested behavior:**

All the global #chatVariables should be configurable while registering @myParticipant in json file. (I feel disabled by default would be good)

So, if a #chatVairable is disabled for @myParticipant, when user uses this participant in prompt the disabled chat variable should not be shown. Making it clear for the user and easier for extension author also avoiding unnecessary prompt processing.

Would be happy to get on a call if above text is not clear.

Thanks.

**Reply by isidorn** [reply]
Understood. Well it is up to your participant to resolve a variable, or remove it from a prompt before you forward that prompt to the LLM. So you are in control.

We have no plans to change this.

fyi @roblourens on user feedback

**Reply by roblourens** [reply]
Why do you want to disable variables rather than using them?

**Reply by amitjoshi438** [reply]
Hi @roblourens 
Based on scenario requirement, we might need to limit the context user is adding using variables.

**Reply by roblourens** [reply]
Why? What's the requirement?

**Reply by amitjoshi438** [reply]
Passing a variable with context will require handling of that context on meta-prompt (LLM side). For our scenario we might not want to support adding a file using #file or terminal info using #terminalSelection as it might lead to in-accurate responses because we use a classifier to identify the user prompt and extra info if not interpreted correctly by LLM will lead to wrong classification. I agree that having more context is helpful in providing accurate response but providing participant that flexibility to choose the context support will be helpful.

As Isidor mentioned we can remove that context before sending to LLM but I believe for a user it can be confusing as user added it in prompt but participant removed it.

**Reply by roblourens** [reply]
You could skip the context for the meta-prompt, so you can focus on the intent, but use it in the real prompt? The context can be helpful for determining the intent too. I'm not really seeing why extra context is not relevant for your participant's scenario.

**Reply by amitjoshi438** [reply]
Thanks, I'll revisit it.

**Comment by theReynald** [comment]
@amitjoshi438, can you respond to roblourens?
